post-til-private/src/main/java/no/difi/meldingsutveksling/dpi/MeldingsformidlerClient.java

```diff
@@ -84,97 +64,56 @@ private Dokument dokumentFromDocument(Document document) {
-            Enumeration<String> aliases = null;
-            try {
-                aliases = trustedSDP.aliases();
-                while (aliases.hasMoreElements()) {
-                    String alias = aliases.nextElement();
-                    Certificate certificate = trustedSDP.getCertificate(alias);
-                    trustStore.setCertificateEntry(alias, certificate);
-                }
-            } catch (KeyStoreException e) {
-                throw new MeldingsUtvekslingRuntimeException("Could not get SDP truststore aliases", e);
-            }
-
-            NoekkelparOverride noekkelparOverride = new NoekkelparOverride(keyStore, trustStore, config.getKeystore().getAlias(), config.getKeystore().getPassword(), false);
-            tekniskAvsender = Databehandler.builder(aktoerOrganisasjonsnummer.forfremTilDatabehandler(), noekkelparOverride).build();
-        } else {
-            tekniskAvsender = Databehandler.builder(aktoerOrganisasjonsnummer.forfremTilDatabehandler(), Noekkelpar.fraKeyStoreUtenTrustStore(keyStore, config.getKeystore().getAlias(), config.getKeystore().getPassword())).build();
-        }
-
-        return new SikkerDigitalPostKlient(tekniskAvsender, klientKonfigurasjon);
-    }
+    public Optional<ExternalReceipt> sjekkEtterKvittering(String orgnr) {
+        Kvittering kvittering = new Kvittering();
 
-    private KlientKonfigurasjon.Builder createKlientKonfigurasjonBuilder() {
-        return KlientKonfigurasjon.builder(config.getEndpoint()).connectionTimeout(20, TimeUnit.SECONDS);
+        PayloadInterceptor payloadInterceptor = new PayloadInterceptor(kvittering::setRawReceipt);
+        SikkerDigitalPostKlient klient = sikkerDigitalPostKlientFactory.createSikkerDigitalPostKlient(AktoerOrganisasjonsnummer.of(orgnr), payloadInterceptor);
+        return getForretningsKvittering(klient)
+                .map(forretningsKvittering -> kvittering.setEksternKvittering(forretningsKvittering)
+                        .withCallback(klient::bekreft)
+                );
     }
 
-
-    public ExternalReceipt sjekkEtterKvittering(String orgnr) {
-        Kvittering kvittering = new Kvittering();
-        PayloadInterceptor payloadInterceptor = new PayloadInterceptor(kvittering::setRawReceipt);
-        SikkerDigitalPostKlient klient = createSikkerDigitalPostKlient(AktoerOrganisasjonsnummer.of(orgnr), payloadInterceptor);
-        final ForretningsKvittering forretningsKvittering = klient.hentKvittering(KvitteringForespoersel.builder(config.getPriority()).mpcId(config.getMpcId()).build());
-        if (forretningsKvittering == null) {
-            return EMPTY_KVITTERING;
+    private Optional<ForretningsKvittering> getForretningsKvittering(SikkerDigitalPostKlient klient) {
+        try {
+            return Optional.ofNullable(klient.hentKvittering(KvitteringForespoersel.builder(config.getPriority()).mpcId(config.getMpcId()).build()));
+        } catch (SendException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)